src/protos/io/defang/v1/fabric.proto

```diff
@@ -96,6 +96,19 @@ service FabricController {
   rpc DeleteMe(google.protobuf.Empty) returns (google.protobuf.Empty) {
     option idempotency_level = IDEMPOTENT;
   }
+
+  rpc VerifyDNSSetup(VerifyDNSSetupRequest) returns (VerifyDNSSetupResponse) {
+    option idempotency_level = NO_SIDE_EFFECTS;
+  }
+}
+
+message VerifyDNSSetupRequest {
+  string domain = 1;
+  string target = 2;
+}
+
+message VerifyDNSSetupResponse {
+  bool success = 1;
```

Comment:
I'd skip this (return `Empty`?) and use the existing gRPC `FailedPrecondition` error code.